my-app/src/redux/game/reducers.js

```diff
@@ -0,0 +1,43 @@
+import calculateWinner from '../../app/utils//utils';
+
+import { actions } from './actions';
+
+const initialState = {
+  history: [{ squares: Array(9).fill(null) }],
+  xIsNext: true,
+  stepNumber: 0
+};
+
+function reducer(state = initialState, action) {
+  switch (action.type) {
+    case actions.handleClick: {
+      const historyCopy = state.history.slice(0, state.stepNumber + 1);
+      const current = historyCopy[historyCopy.length - 1];
+      const squares = current.squares.slice();
+      if (calculateWinner(squares) || squares[action.pos]) {
+        return state;
+      }
+      squares[action.pos] = state.xIsNext ? 'X' : 'O';
+      return {
+        history: historyCopy.concat([
```

Comment:
We can use `history` directly without making it a copy, since `concat` returns a new array. What do you think ?

Reply:
that's true, it will be fixed on the next commit :)